test/e2e/app-dir/navigation/navigation.test.ts

```diff
@@ -585,21 +585,30 @@ describe('app dir - navigation', () => {
     })
 
     describe('status code', () => {
-      it('should respond with 307 status code in server component', async () => {
+      it('should respond with 307 status code in server component for html bots', async () => {
         const res = await next.fetch('/redirect/servercomponent', {
           redirect: 'manual',
+          headers: {
+            'User-Agent': 'Discordbot',
+          },
```

Comment:
i wouldn't expect this to be necessary for these routes, I don't think anything would stream here as these pages just return regular JSX

Reply:
Since the metadata resloving itself is suspended, the status will become always 200. It won't resolved on server side, hence I changed it to bot request.

The logic of resolving notFound metadata is not solid yet, usually if you do `notFound` you'll need both logic in `generateMetadata` and the page itself, you actually don't have to do `notFound` in `generateMetadata`.

Reply:
in this case tho, `/redirect/servercomponent` doesn't even use metadata. Why would it need a change to the user agent?

Reply:
right they're not necessary, messed up with the metadata tests before